Modules/_pickle.c

```diff
@@ -5126,42 +5112,39 @@ find_class(UnpicklerObject *self, PyObject *module_name, PyObject *global_name)
 }
 
 static Py_ssize_t
-marker(UnpicklerObject *self)
+marker(UnpicklerObject *self, PickleState *st)
 {
-    Py_ssize_t mark;
-
     if (self->num_marks < 1) {
-        PickleState *st = _Pickle_GetGlobalState();
         PyErr_SetString(st->UnpicklingError, "could not find MARK");
         return -1;
     }
 
-    mark = self->marks[--self->num_marks];
+    Py_ssize_t mark = self->marks[--self->num_marks];
     self->stack->mark_set = self->num_marks != 0;
     self->stack->fence = self->num_marks ?
             self->marks[self->num_marks - 1] : 0;
     return mark;
 }
 
 static int
-load_none(UnpicklerObject *self)
+load_none(UnpicklerObject *self, PickleState *st)
```

Comment:
PickleState is not used here, but is added as an argument to avoid complicating the macro/switch case in 
static PyObject * load(UnpicklerObject *self, PickleState *st)